File: crates/taflow-core/src/indicators/hilbert_transform_dominant_cycle_phase_test.rs
use super::hilbert_transform_dominant_cycle_phase::HilbertTransformDominantCyclePhase;

#[test]
fn lifecycle_and_reset_are_causal() {
    let mut state = HilbertTransformDominantCyclePhase::new();
    for value in (0..100).map(|i| 100.0 + (i as f64 * 0.1).sin()) {
        state.append(value);
    }
    assert!(state.value().is_some());
    state.reset();
    assert_eq!(state.value(), None);
}

File: tests/hilbert_transform_dominant_cycle_phase_test.py
import numpy as np

from taflow import HilbertTransformDominantCyclePhase


def test_hilbert_transform_dominant_cycle_phase_lifecycle():
    values = np.sin(np.linspace(0.0, 10.0, 100))
    indicator = HilbertTransformDominantCyclePhase().extend(values)
    assert len(indicator) == len(values)
    indicator.reset()
    assert indicator.value is None
